### PR TITLE
Added a doc for IDE setup

### PR DESCRIPTION
**Pull request by Samyoul**
Based on https://notes.status.im/9Gr7kqF8SzC_SmYK0eB7uQ?view, hopefully this document can help some poor soul in the future. I have a little more work to do but I'm interested in whether this kind of document would be welcome in the repo.

After `make clean` I found that I had a lot of reset up to do and it took me a long time to find all the information to recreate the setup. This document will be invaluable for at least a future me, so maybe it could be useful to others.

To easily see the document see https://github.com/status-im/status-react/blob/55ccda27340f97f874c623abd3c9072989ccd199/doc/IDE_SETUP.md

---

Todos for this PR:

- [x] Add a table of contents
- [x] Tidy up the Troubleshooting parts
  - [x] Remove references to things I tried and detail from a this could go wrong, this is how to fix it perspective
- [x] Add all images to the repo to prevent future 404s
- [x] Some images are huge others are small, resize big images so the document is easier to read
- [x] Add more detailed description of connecting and using REPL with IntelliJ
- [x] Link to other parts in the docs that are relevant

### CODE REVIEW

**Review by shivekkhurana — APPROVED** [review]

**Comment by flexsurfer** [comment]
i guess its WIP ?

**Comment by Samyoul** [comment]
Yes let me rename it

File: doc/IDE_SETUP.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+prn 1
+```
+
+Which should output
+
+```shell
+prn 1
+=> 1
+=> #function[clojure.core/prn]
+```
```
Comment (shivekkhurana) [inline doc/IDE_SETUP.md]: This doesn't sound right `prn 1` is not valid without the parenthesis

File: doc/IDE_SETUP.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+Obviously this isn't ideal even if it worked perfectly, I presume that the file types are important and converting them back and forth is a terrible work flow.
+
+I tried converting the files to `clj`, just to see if that helped, which gave an interesting error:
+
+```bash=
+Loading src/status_im/reloader.clj... 
+Syntax error (FileNotFoundException) compiling at (src/status_im/reloader.clj:1:1).
+Could not locate status_im/ui/components/react__init.class, status_im/ui/components/react.clj or status_im/ui/components/react.cljc on classpath. Please check that namespaces with dashes use underscores in the Clojure file name.
+```
```
Comment (shivekkhurana) [inline doc/IDE_SETUP.md]: Are these two sections needed since the sections above show how to connect to the repl?

**Review by shivekkhurana — CHANGES_REQUESTED** [review]
Some sections can be removed.

File: doc/IDE_SETUP.md (hunk 1 above)
Comment (Samyoul) [reply doc/IDE_SETUP.md]: It works on my machine, you can see in the screenshot below. But it is a bit redundant so I've removed it from the docs

File: doc/IDE_SETUP.md (hunk 2 above)
Comment (Samyoul) [reply doc/IDE_SETUP.md]: Yeah, I've removed this section

**Review by flexsurfer — APPROVED** [review]

**Review by shivekkhurana — APPROVED** [review]

**Comment by shivekkhurana** [comment]
@Samyoul Please feel free to merge this (we don't use github's merge functionality, but the [merge script instead](https://github.com/status-im/status-react/tree/develop/doc#merging-approved-prs)).

**Comment by Samyoul** [comment]
That's an interesting merge process.

**Pull request merged**